mmv1/third_party/terraform/services/composer/resource_composer_environment.go.erb

```diff
@@ -2922,7 +3020,45 @@ func versionsEqual(old, new string) (bool, error) {
 	return o.Equal(n), nil
 }
 
-func isComposer3(d *schema.ResourceData, config *transport_tpg.Config)	bool {
-	image_version := d.Get("config.0.software_config.0.image_version").(string)
-	return strings.Contains(image_version, "composer-3")
+func isComposer3(d interface{})	bool {
+	var imageVersion string
+	switch d.(type) {
+	case *schema.ResourceData:
+		imageVersion= d.(*schema.ResourceData).Get("config.0.software_config.0.image_version").(string)
+	case *schema.ResourceDiff:
+		imageVersion= d.(*schema.ResourceDiff).Get("config.0.software_config.0.image_version").(string)
+	default:
+		log.Printf("[ERROR] error in plugin: unexpected argument type in isComposer3")
+	}
+	return strings.Contains(imageVersion, "composer-3")
+}
+
+func forceNewCustomDiff(key string) customdiff.ResourceConditionFunc {
+	return func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+		old, new := d.GetChange(key)
+		if isComposer3(d) || networkReferenceEquivalent(old.(string), new.(string)) {
+			return false
+		}
+		return true
+	}
 }
+
+// Checks if two network/subnetwork references are equivalent
+func networkReferenceEquivalent(s1, s2 string) bool {
```

Comment:
We can use a function from https://github.com/hashicorp/terraform-provider-google-beta/blob/main/google-beta/tpgresource/self_link_helpers.go#L22 instead, that'll avoid needing to bake a GCE url into the file.

Reply:
Implemented this change. Thanks for the suggestion.